mesonbuild/compilers/java.py

```diff
@@ -83,10 +83,10 @@ def get_buildtype_args(self, buildtype):
 
     def compute_parameters_with_absolute_paths(self, parameter_list, build_dir):
         for idx, i in enumerate(parameter_list):
-            if i[:4] == '-cp:' or i[:4] == '-cp;':
-                parameter_list[idx] = i[:4] + os.path.normpath(os.path.join(build_dir, i[4:]))
-            if i[:11] == '-classpath:' or i[:11] == '-classpath;':
-                parameter_list[idx] = i[:11] + os.path.normpath(os.path.join(build_dir, i[11:]))
+            if i in ['-cp', '-classpath', '-sourcepath'] and idx + 1 < len(parameter_list):
+                path_list = parameter_list[idx + 1].replace(';', ':').split(':')
```

Comment:
We have to use os.pathsep while splitting and joining. Because windows. Changing ['C:\blah'] to ['C', '\blah'] would break poor windows.